wikimedia/wikipedia-ios
# Add article inspector mapping (part 3 of 4)

**Fixes Phabricator ticket:** https://phabricator.wikimedia.org/T270426

Do not review until #3864 is merged and this PR is updated with main.

### Notes
* This doesn't complete the Phabricator task - this is the third step that links the separated sentences from the article content and WikiWho response and joints them together in to a `CombinedSentence` struct. Final processing to generate NSAttributedStrings out of this for display in the modal will be in a followup PR.

### Test Steps

1. Set isEnabled = true in ArticleInspectorController
2. Add a breakpoint to the processContent(articleHtml: String, wikiWhoResponse: WikiWhoResponse) method, after the combinedSections are created.
2. Run and go to the article that is cached on the endpoint. You might need to visit https://wikiwho-ios-experiments.wmflabs.org/whocolor/{article_title} to kick off the caching. See that combinedSections are properly populated at the breakpoint (note I also added some light unit tests that we can flesh out more fully as we run into issues). You should now see CombinedSentence structs within that contain a.) the `rawText` - a single sentence without html tags b.) the `articleText` - the sentence we will be seeking out and highlighting in the article content across the JS bridge and c.) the `nativeText` - or what we will be using to generate NSAttributedStrings from for modal display in the next PR.
